pkg/providers/v1/config/config.go

```diff
@@ -83,6 +83,16 @@ type CloudConfig struct {
 
 		// ClusterServiceSharedLoadBalancerHealthProbePath defines the target path of the shared health probe. Default to `/healthz`.
 		ClusterServiceSharedLoadBalancerHealthProbePath string `json:"clusterServiceSharedLoadBalancerHealthProbePath,omitempty" yaml:"clusterServiceSharedLoadBalancerHealthProbePath,omitempty"`
+
+		// Override to regex validating whether or not instance types require instance topology
+		// to get a definitive response. This will impact whether or not the node controller will
+		// block on getting instance topology information for nodes. Use cases to modify this:
+		// 1. You don't care about node topology labels and want nodes to succeed ASAP.
+		// 2. You want to ensure that a new instance types requires a response.
+		// See pkg/resourcemanagers/topology.go for more details.
+		//
+		// WARNING: Updating the default behavior and corresponding unit tests would be a much safer option.
+		InstanceTypePatternRequireTopologyResponseOverride string `json:"instanceTypePatternRequireTopologyResponseOverride,omitempty" yaml:"instanceTypePatternRequireTopologyResponseOverride,omitempty"`
```

Comment:
Chatted offline, maybe name `supportedTopologyInstanceTypePattern` to be clearer.

Ideally we could use the standard defaulter-gen logic to set the default regex, but we're not using any of the proper API machinery, so not worth doing in this PR.